includes/Modules/Analytics_4/Web_Tag.php

```diff
@@ -150,27 +149,31 @@ protected function setup_gtag( GTag $gtag ) {
 		if ( $this->ads_conversion_id ) {
 			$gtag->add_tag( $this->ads_conversion_id );
 		}
+	}
 
-		$filter_google_gtagjs = function ( $tag, $handle ) use ( $gtag ) {
-			if ( GTag::HANDLE !== $handle ) {
-				return $tag;
-			}
-
-			// Retain this comment for detection of Site Kit placed tag.
-			$snippet_comment = sprintf( "\n<!-- %s -->\n", esc_html__( 'Google Analytics snippet added by Site Kit', 'google-site-kit' ) );
-
-			$block_on_consent_attrs = $this->get_tag_blocked_on_consent_attribute();
+	/**
+	 * Filters output of tag HTML.
+	 *
+	 * @param string $tag Tag HTML.
+	 * @param string $handle WP script handle of given tag.
+	 * @return string
+	 */
+	protected function filter_tag_output( $tag, $handle ) {
+		// The tag will either have its own handle or use the common GTag handle, not both.
+		if ( GTag::get_handle_for_tag( $this->tag_id ) !== $handle && GTag::HANDLE !== $handle ) {
+			return $tag;
+		}
 
-			if ( $block_on_consent_attrs ) {
-				$gtag_src = $gtag->get_gtag_src();
+		// Retain this comment for detection of Site Kit placed tag.
+		$snippet_comment = sprintf( "<!-- %s -->\n", esc_html__( 'Google Analytics snippet added by Site Kit', 'google-site-kit' ) );
 
-				$tag = $this->add_legacy_block_on_consent_attributes( $tag, $gtag_src, $block_on_consent_attrs );
-			}
+		$block_on_consent_attrs = $this->get_tag_blocked_on_consent_attribute();
```

Comment:
Let's use the whole word here (and below) 😅 

```suggestion
		$block_on_consent_attributes = $this->get_tag_blocked_on_consent_attribute();
```